testsuite/testcases/src/network_chaos_test.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::generate_traffic;
+use forge::{NetworkContext, NetworkTest, Result, SwarmChaos, Test};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+pub trait NetworkChaosTest: Test {
+    fn get_chaos(&self) -> SwarmChaos;
+    fn get_message(&self) -> String;
+}
+
+impl NetworkTest for dyn NetworkChaosTest {
```

Comment:
you might want to extend NetworkLoadTest, I don't see anything here that is too special for chaos tests.

you would override setup() to inject_chaos, and you can move finish (that you removed) to be before check_for_success, so that you can call remove_chaos() there

Reply:
Good point, done.